File: src/go/data_processor_004.go
package main

import (
	"fmt"
	"strings"
	"time"
)

type DataItem struct {
	ID    int    `json:"id"`
	Name  string `json:"name"`
	Value int    `json:"value"`
}

type ProcessedItem struct {
	ID          int       `json:"id"`
	Name        string    `json:"name"`
	Value       int       `json:"value"`
	ProcessedAt time.Time `json:"processed_at"`
}

type DataProcessor004 struct {
	config        map[string]interface{}
	data          []DataItem
	processedCount int
}

func NewDataProcessor004(config map[string]interface{}) *DataProcessor004 {
	return &DataProcessor004{
		config:        config,
		data:          make([]DataItem, 0),
		processedCount: 0,
	}
}

func (dp *DataProcessor004) ProcessData(items []DataItem) map[int]ProcessedItem {
	results := make(map[int]ProcessedItem)
	
	for _, item := range items {
		if dp.validateItem(item) {
			processed := dp.transformItem(item)
			results[item.ID] = processed
			dp.processedCount++
		}
	}
	
	return results
}

func (dp *DataProcessor004) validateItem(item DataItem) bool {
	return item.ID > 0 && item.Name != "" && item.Value > 0
}

func (dp *DataProcessor004) transformItem(item DataItem) ProcessedItem {
	return ProcessedItem{
		ID:          item.ID,
		Name:        strings.ToUpper(item.Name),
		Value:       item.Value * 4,
		ProcessedAt: time.Now(),
	}
}

func (dp *DataProcessor004) GetStatistics() map[string]interface{} {
	return map[string]interface{}{
		"processed_count": dp.processedCount,
		"config_keys":     len(dp.config),
		"data_size":       len(dp.data),
	}
}

File: src/java/DataProcessor004.java
import java.util.*;
import java.time.LocalDateTime;
import java.time.format.DateTimeFormatter;

public class DataProcessor004 {
    private Map<String, Object> config;
    private List<Map<String, Object>> data;
    private int processedCount;
    
    public DataProcessor004(Map<String, Object> config) {
        this.config = config;
        this.data = new ArrayList<>();
        this.processedCount = 0;
    }
    
    public Map<String, Object> processData(List<Map<String, Object>> inputData) {
        Map<String, Object> results = new HashMap<>();
        
        for (Map<String, Object> item : inputData) {
            if (validateItem(item)) {
                Map<String, Object> processed = transformItem(item);
                results.put(item.get("id").toString(), processed);
                processedCount++;
            }
        }
        
        return results;
    }
    
    private boolean validateItem(Map<String, Object> item) {
        String[] requiredFields = {"id", "name", "value"};
        for (String field : requiredFields) {
            if (!item.containsKey(field)) {
                return false;
            }
        }
        return true;
    }
    
    private Map<String, Object> transformItem(Map<String, Object> item) {
        Map<String, Object> processed = new HashMap<>();
        processed.put("id", item.get("id"));
        processed.put("name", item.get("name").toString().toUpperCase());
        processed.put("value", ((Number) item.get("value")).intValue() * 4);
        processed.put("processedAt", LocalDateTime.now().format(DateTimeFormatter.ISO_LOCAL_DATE_TIME));
        return processed;
    }
    
    public Map<String, Object> getStatistics() {
        Map<String, Object> stats = new HashMap<>();
        stats.put("processedCount", processedCount);
        stats.put("configKeys", config.size());
        stats.put("dataSize", data.size());
        return stats;
    }
}
